recipe/meta.yaml

```diff
@@ -146,130 +155,114 @@ outputs:
         - sqlite
         - zlib
         # requirements specified by the package itself
-        - absl-py >=0.10,<1
-        - astunparse >=1.6.3,<1.7
-        - python-flatbuffers >=1.12,<2
-        - google-pasta >=0.2,<1
-        - h5py >=3.1.0,<3.2
-        - keras-preprocessing >=1.1.2,<1.2
-        - numpy >=1.19.2,<1.20
-        - opt_einsum >=3.3,<3.4
+        - absl-py >=0.4.0
+        - astunparse >=1.6.0
+        - gast >=0.2.1,<0.5.0
+        - google-pasta >=0.1.1
+        - h5py >=2.9.0
+        - keras-preprocessing >=1.1.1
+        - numpy
+        - opt_einsum >=2.3.2
         - protobuf >=3.9.2
-        - six >=1.15,<1.16
-        - termcolor >=1.1,<1.2
-        - typing_extensions >=3.7.4,<3.8
-        - wheel >=0.35,<1
-        - wrapt >=1.12.1,<1.13
-        # upstream wants exact pin for gast
-        - gast ==0.4.0
+        - python-flatbuffers >=1.12,<3
+        - six >=1.12
+        - tensorflow-io-gcs-filesystem >=0.21.0
+        - termcolor >=1.1.0
+        - typing_extensions >=3.6.6
+        - wheel >=0.32,<1
+        - wrapt >=1.11.0
         # TF-API needs to move in sync
+        # XXX: actually, TF 2.7.0 still demands tb 2.6
         - tensorboard >=2.6,<2.7
-        - keras >=2.6,<2.7
+        - keras >=2.7,<2.8
       run:
         - python
-        - absl-py >=0.10,<1
-        - astunparse >=1.6.3,<1.7
-        - python-flatbuffers >=1.12,<2
-        - google-pasta >=0.2,<1
+        - absl-py >=0.4.0
+        - astunparse >=1.6.0
+        - gast >=0.2.1,<0.5.0
+        - google-pasta >=0.1.1
         - grpcio {{ grpc_cpp }}.*
-        - h5py >=3.1.0,<3.2
-        - keras-preprocessing >=1.1.2,<1.2
-        - numpy >=1.19.2,<1.20
-        - opt_einsum >=3.3,<3.4
+        - h5py >=2.9.0
+        - keras-preprocessing >=1.1.1
+        - {{ pin_compatible('numpy') }}
+        - opt_einsum >=2.3.2
         - protobuf >=3.9.2
-        - six >=1.15,<1.16
-        - termcolor >=1.1,<1.2
-        - typing_extensions >=3.7.4,<3.8
-        - wheel >=0.35,<1
-        - wrapt >=1.12.1,<1.13
-        # upstream wants exact pin for gast
-        - gast ==0.4.0
+        - python-flatbuffers >=1.12,<3
+        - six >=1.12
+        - tensorflow-io-gcs-filesystem >=0.21.0
```

Comment:
I really do not think this is a dependency.

Reply:
Thanks for checking upstream, I did not see that this was conditional.

However, I'm almost tempted to open an issue upstream about the circularity with tensorflow-io (if they make it unconditional, as announced).

Reply:
you should definitely open an issue to raise your concerns.